articles/openshift/oshift-ref-security-best-practices.md

```diff
@@ -0,0 +1,112 @@
+
+## Limiting resources
+
+You should always set up containers with resource limits. In the event of a denial of service attack, or even just a burst of very high load on a service, these resource limits will prevent pods from using all available resources in the node they are running on. You can read more about setting resource quotas and limits [here](https://docs.openshift.com/container-platform/3.11/dev_guide/compute_resources.html).
+
+## Privileged ports
+
+TCP/IP ports below port 1024 are considered to be privileged ports and you should not use them to front any service that is deployed into OpenShift.
+
+## Route security
+
+Within OpenShift, you can expose your services using routes. Creating a route to a service enables your users to access the service through a URL on a browser. By default, the URLs that are created are exposed on HTTP port 80 without any SSL certificate. You can secure these routes with SSL encryption using one of three SSL termination methods:
+
+ - Edge - SSL termination is done on the HAProxy router from which point onwards the traffic is un-encrypted
+
+ - Passthrough termination - There is no SSL decryption on the routers; instead connections are passed straight through the routers to the OpenShift applications, which are then responsible for SSL certificate termination and traffic decryption.
+
+ - Re-Encrypt - This process uses two certificates: the first certificate is held on the HAProxy router and the second in the pods. This secures traffic all the way down to the application.
+
+It is possible to further secure routes by applying IP whitelisting on routes. We have documented the process of IP whitelisting [here](oshift-how-restrict-access-to-openshift-routes-by-ip-address.md).
+
+UKCloud can whitelist individual IP addresses or entire subnets to access the OpenShift console and metrics page. It is also possible to do the same for the services that you create and deploy in OpenShift, even if those services are exposed to various Government Community Networks with the use of security groups. These restrictions can be applied at the time the cluster is deployed or can be done later via a Service Request to the UKCloud OpenShift team.
+
+## Software-defined networking (SDN)
+
+OpenShift uses a software-defined networking (SDN) approach to provide cluster-wide networking that enables communication between pods across the OpenShift cluster. This Container Networking Interface (CNI) runs in the OpenShift SDN project and creates an overlay network using the Open vSwitch (OVS) networking plugin.
+
+However, from February 2019, UKCloud will be deploying OpenShift clusters with the ovs-networkpolicy plugin. This plugin allows project administrators to configure their own isolation policies using [NetworkPolicy objects](https://docs.openshift.com/container-platform/3.11/admin_guide/managing_networking.html#admin-guide-networking-networkpolicy). You can find an article regarding NetworkPolicy [here](oshift-netpol.md).
```

Comment:
This reads to me like we're not going to be using ovs in the network policy SDN, but I think it still does. Perhaps removing the however will fix that. I think we should remove the date reference as well and just state in earlier versions we deployed with ovs-multitenant, and now we deploy with ovs-networkpolicy and view this article to find out more, but basically one has more granular control but is easier to open access inadvertently.